src/main/scala/com/microsoft/hyperspace/actions/Action.scala

```diff
@@ -84,18 +84,21 @@ trait Action extends HyperspaceEventLogging with Logging with ActiveSparkSession
     val appInfo =
       AppInfo(sparkContext.sparkUser, sparkContext.applicationId, sparkContext.appName)
     try {
-      logEvent(event(appInfo, "Operation Started."))
+      logEvent(event(appInfo, "Operation started."))
       validate()
 
       begin()
 
       op()
 
       end()
-      logEvent(event(appInfo, message = "Operation Succeeded."))
+      logEvent(event(appInfo, message = "Operation succeeded."))
     } catch {
+      case e: NoChangesException =>
+        logEvent(event(appInfo, message = s"No-op operation recorded: ${e.getMessage}"))
+        logWarning(e.msg)
       case e: Exception =>
-        logEvent(event(appInfo, message = s"Operation Failed: ${e.getMessage}."))
+        logEvent(event(appInfo, message = s"Operation failed: ${e.getMessage}"))
```

Comment:
for reviewers: the trailing full stop "." is removed in the lines after `${e.getMessage}` because there are cases when the `e.getMessage` contains a `.` (refer [this](https://github.com/microsoft/hyperspace/pull/189/files#diff-14be32100e14d26db8bc2ed7a662b1e8R62) for example)